service/src/main/scala/za/co/absa/loginsvc/rest/config/jwt/AwsSecretsManagerKeyConfig.scala

```diff
@@ -0,0 +1,115 @@
+                                       region: String,
+                                       privateKeyFieldName: String,
+                                       publicKeyFieldName: String,
+                                       algName: String,
+                                       accessExpTime: FiniteDuration,
+                                       pollTime: Option[FiniteDuration])
+  extends KeyConfig {
+
+  private val logger = LoggerFactory.getLogger(classOf[AwsSecretsManagerKeyConfig])
+
+  override def refreshKeyTime : Option[FiniteDuration] = pollTime
+  override def keyPair(): KeyPair = {
+
+    val default = DefaultCredentialsProvider.create
+
+    val client = SecretsManagerClient.builder
+      .region(Region.of(region))
+      .credentialsProvider(default)
+      .build
+
+    val getSecretValueRequest = GetSecretValueRequest.builder.secretId(secretName).build
+
+    try {
+      logger.info("Attempting to fetch key data from AWS Secrets Manager")
+      val getSecretValueResponse: GetSecretValueResponse = client.getSecretValue(getSecretValueRequest)
+      val secret = getSecretValueResponse.secretString
+      logger.info("Key data retrieved. Attempting to Parse key data")
+      val rootNode: JsonNode = new ObjectMapper().readTree(secret)
+
+      val publicKeySpec: X509EncodedKeySpec = new X509EncodedKeySpec(
+        Base64.getDecoder.decode(
+          rootNode.get(publicKeyFieldName).asText()
+        )
+      )
+      val privateKeySpec: PKCS8EncodedKeySpec = new PKCS8EncodedKeySpec(
+        Base64.getDecoder.decode(
+          rootNode.get(privateKeyFieldName).asText()
+        )
+      )
+
+      logger.info("Key Data successfully retrieved and parsed from AWS Secrets Manager")
+
+      val keyFactory: KeyFactory = KeyFactory.getInstance(jwtAlgorithmToCryptoAlgorithm)
+      new KeyPair(keyFactory.generatePublic(publicKeySpec), keyFactory.generatePrivate(privateKeySpec))
+    }
+    catch {
+      case e: Throwable =>
+        logger.error(s"Error occurred retrieving and decoding keys from AWS Secrets Manager", e)
+        throw e
+    }
+  }
+  override def throwErrors(): Unit = this.validate().throwOnErrors()
+
+  override def validate(): ConfigValidationResult = {
+
+    val defaultResults = defaultValidation
+
+    val awsSecretsResults = Seq(
+      Option(secretName)
+        .map(_ => ConfigValidationSuccess)
+        .getOrElse(ConfigValidationError(ConfigValidationException("secretName is empty"))),
+
+      Option(region)
+        .map(_ => ConfigValidationSuccess)
+        .getOrElse(ConfigValidationError(ConfigValidationException("region is empty"))),
+
+      Option(privateKeyFieldName)
+        .map(_ => ConfigValidationSuccess)
+        .getOrElse(ConfigValidationError(ConfigValidationException("privateAwsKey is empty"))),
+
+      Option(publicKeyFieldName)
+        .map(_ => ConfigValidationSuccess)
+        .getOrElse(ConfigValidationError(ConfigValidationException("publicAwsKey is empty"))),
```

Comment:
```suggestion
      Option(privateKeyFieldName)
        .map(_ => ConfigValidationSuccess)
        .getOrElse(ConfigValidationError(ConfigValidationException("privateKeyFieldName is empty"))),

      Option(publicKeyFieldName)
        .map(_ => ConfigValidationSuccess)
        .getOrElse(ConfigValidationError(ConfigValidationException("publicKeyFieldName is empty"))),
```

Thanks, but you forgot to change the exception message.

Reply:
Implemented